edb/edgeql/compiler/setgen.py

```diff
@@ -298,6 +298,13 @@ def compile_path(expr: qlast.Path, *, ctx: context.ContextLevel) -> irast.Set:
 
                 if (stype.get_expr_type(ctx.env.schema) is not None and
                         stype.get_name(ctx.env.schema) not in ctx.view_nodes):
+                    if not stype.get_expr(ctx.env.schema):
+                        raise errors.InvalidReferenceError(
+                            f"cannot refer to alias link helper type "
```

Comment:
Should we pretend it doesn't exist instead?

Reply:
I considered that, but it's visible in schema reflection, and if anybody is trying to select it they probably saw it in schema reflection, so this is a more useful error?

I don't feel strongly though

Reply:
OK, fine by me.